contracts/Strategy.sol

```diff
@@ -17,95 +17,389 @@ import {
+     *  losses for the Vault based on sudden withdrawals. This value should be
+     *  higher than the total debt of the Strategy and higher than its expected
+     *  value to be "safe".
+     * @return The estimated total assets in this Strategy.
+     */
+    function estimatedTotalAssets() public override view returns (uint256){
+        (uint256 _mToken, ) = MMFarmingPool(mmFarmingPool).userInfo(mmFarmingPoolId, address(this));
+        uint256 _mmVault = IERC20(mmVault).balanceOf(address(this));
+        return _convertMTokenToWant(_mToken.add(_mmVault));
     }
 
+    /**
+     * Perform any Strategy unwinding or other calls necessary to capture the
+     * "free return" this Strategy has generated since the last time its core
+     * position(s) were adjusted. Examples include unwrapping extra rewards.
+     * This call is only used during "normal operation" of a Strategy, and
+     * should be optimized to minimize losses as much as possible.
+     *
+     * This method returns any realized profits and/or realized losses
+     * incurred, and should return the total amounts of profits/losses/debt
+     * payments (in `want` tokens) for the Vault's accounting (e.g.
+     * `want.balanceOf(this) >= _debtPayment + _profit - _loss`).
+     *
+     * `_debtOutstanding` will be 0 if the Strategy is not past the configured
+     * debt limit, otherwise its value will be how far past the debt limit
+     * the Strategy is. The Strategy's debt limit is configured in the Vault.
+     *
+     * NOTE: `_debtPayment` should be less than or equal to `_debtOutstanding`.
+     *       It is okay for it to be less than `_debtOutstanding`, as that
+     *       should only used as a guide for how much is left to pay back.
+     *       Payments should be made to minimize loss from slippage, debt,
+     *       withdrawal fees, etc.
+     *
+     * See `vault.debtOutstanding()`.
+     */
     function prepareReturn(uint256 _debtOutstanding)
         internal
         override
         returns (
             uint256 _profit,
             uint256 _loss,
             uint256 _debtPayment
-        )
-    {
-        // TODO: Do stuff here to free up any returns back into `want`
-        // NOTE: Return `_profit` which is value generated by all positions, priced in `want`
-        // NOTE: Should try to free up at least `_debtOutstanding` of underlying position
+    ){
+        _profit = 0;
+        _loss = 0; //for clarity. also reduces bytesize
+        _debtPayment = 0; //for clarity. also reduces bytesize
+		
+        // Pay debt if any
+        if (_debtOutstanding > 0) {
+            (uint256 _amountFreed, uint256 _reportLoss) = liquidatePosition(_debtOutstanding);
+            _debtPayment = _amountFreed > _debtOutstanding? _debtOutstanding : _amountFreed;
+            _loss = _reportLoss;
+        }
+		
+        // Claim profit
+        MMFarmingPool(mmFarmingPool).withdraw(mmFarmingPoolId, 0);
```

Comment:
isn't there a way to check if there is profit available before calling `withdraw`?